Get rid of newWithSize.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34403 Port set_ to ATen.
* #34389 Stop creating unnecessary Storage with newWithStorage1d.
* #34388 Kill newWithStorage.
* **#34387 Get rid of newWithSize.**

Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Get rid of newWithSize.

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

bhosmer commented on aten/src/THC/generic/THCTensor.cpp:
```diff
@@ -100,20 +100,21 @@ THCTensor *THCTensor_(newWithStorage1d)(THCState *state, THCStorage *storage, pt
+    c10::intrusive_ptr<at::StorageImpl>::reclaim(new_storage),
+    at::DispatchKey::CUDATensorId
+  ).release();
+  THCTensor_(setStorageNd)(state, self, 0, 1, &size0, nullptr);
```
Another pattern-match observation, `THTensor` version passes `new_storage` rather than `self` as second arg

gchanan replied:
good catch, although the "second arg" thing is because THCTensor always have extra state.  But I was indeed neglecting to pass new_storage.

bhosmer reviewed (approved):
See inline for possible thing

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)

gchanan committed:
Update on "Get rid of newWithSize."


Differential Revision: [D20311058](https://our.internmc.facebook.com/intern/diff/D20311058)
